src/main/java/com/iexec/common/utils/purge/ExpiringTaskMapFactory.java

```diff
@@ -0,0 +1,25 @@
+package com.iexec.common.utils.purge;
+
+import net.jodah.expiringmap.ExpiringMap;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An expiring task map is defined as a map of {@link String} as keys
+ * and any type as keys,
+ * whereas the map can be purged after a given amount of time -
+ * max duration of any task, currently of 100 hours.
+ */
+public class ExpiringTaskMapFactory {
+    private ExpiringTaskMapFactory() {
+
+    }
+
+    public static <T> Map<String, T> getExpiringTaskMap() {
+        return ExpiringMap
+                .builder()
+                .expiration(100, TimeUnit.HOURS)
```

Comment:
hmm, I understand this should be a big failsafe to avoid memory leak.
Could the values not be hardcoded and be extracted from the blockchain or be explicitly related to biggest deadlines in categories ?

Reply:
As of now, we can't easily extract the value from the blockchain. So let's store it as a constant and remind ourselves this could be something to update for PoCo v2. Please see https://github.com/iExecBlockchainComputing/iexec-common/pull/355/commits/80237f7820905618b2a8eeafe616c6e41351c6d3 :)